### PR TITLE
Added options as a 3rd parameter and onError callback

### PR DESCRIPTION
**Pull request by danikaze**
If there's an error while creating the webgl context, **GlslCanvas** will automatically set the contents of the canvas with a global impact.
This option allows the user to control and handle the _error_ manually with a callback.

```
function onError(errorCode) {
  // set some flags in the application to continue without webgl support
}

const glsl = new GlslCanvas(canvas, null, { onError });
```

It's added as a new `options` parameter to keep the 2nd parameter clean with only `contextOptions` (**WebGLContextCreationAttributes** type)

### CODE REVIEW

**Comment by patriciogonzalezvivo** [comment]
awesome! is this 3er parameter optional?

**Comment by danikaze** [comment]
Thanks
Sure it is! It defaults to `{}` (in `src/glslCanvas.js` constructor).

Right now there's only the `onError` options, but I thought it's better to have it as an object so if there's any new option it can be added easily, instead of keep adding more and more parameters in the constructor.

Also, I didn't added `onError` in the 2nd parameter, because I understand it's a WebGLContextCreationAttributes type, so I think it's better not to _pollute_ it with extra properties.

**Comment by patriciogonzalezvivo** [comment]
thanks!

**Comment by danikaze** [comment]
Awesome, prompt and quick action.
10/10 Would commit again lol

File: src/GlslCanvas.js [hunk 1]
```diff
@@ -46,7 +47,7 @@ export default class GlslCanvas {
         this.vbo = {};
         this.isValid = false;
 
-        this.vertexString = options.vertexString || `
+        this.vertexString = contextOptions.vertexString || `
```
Comment (wontem) [inline src/GlslCanvas.js]: `contextOptions` is not suitable here. `contextOptions` is used as the second argument in `canvas.getContext` so it seems that the `options` should be used here. The only concern is the backward compatibility - I suggest to swap `contextOptions` and `options` in the arguments.

File: src/GlslCanvas.js [hunk 2]
```diff
@@ -85,7 +86,7 @@ void main(){
         this.paused = false;
 
         // Allow alpha
-        canvas.style.backgroundColor = options.backgroundColor || 'rgba(1,1,1,0)';
+        canvas.style.backgroundColor = contextOptions.backgroundColor || 'rgba(1,1,1,0)';
```
Comment (wontem) [inline src/GlslCanvas.js]: As above `options` should be here.

File: src/GlslCanvas.js [hunk 3]
```diff
@@ -61,7 +62,7 @@ void main() {
     v_texcoord = a_texcoord;
 }
 `;
-        this.fragmentString = options.fragmentString || `
+        this.fragmentString = contextOptions.fragmentString || `
```
Comment (wontem) [inline src/GlslCanvas.js]: As above `options` should be here.

**Pull request merged**